networking_cisco/plugins/cisco/db/l3/l3_router_appliance_db.py

```diff
@@ -233,12 +238,22 @@ def create_router(self, context, router):
         if auto_schedule is True:
             # backlog so this new router gets scheduled asynchronously
             self.backlog_router(context, r_hd_b_db)
+        if driver:
+            driver.create_router_postcommit(context, router_ctxt)
         return router_created
 
     def update_router(self, context, id, router):
+        router_type_id = self.get_router_type_id(context, id)
+        driver = self._get_router_type_driver(context,
+                                              router_type_id)
+        if driver:
+            router_ctxt = driver_context.RouterContext(router)
+            driver.update_router_precommit(context, router_ctxt)
         router_updated = self._update_router_no_notify(context, id, router)
         self.add_type_and_hosting_device_info(context.elevated(),
                                               router_updated)
+        driver = self._get_router_type_driver(context,
```

Comment:
Actually, correction -- the router will be the same in pre- and post- for this API call.